File: Xamarin/LightOrganApp.iOS/CustomSearchController.cs
using CoreGraphics;
using Foundation;
using System;
using System.Collections.Generic;
using System.Text;
using UIKit;

namespace LightOrganApp.iOS
{
    public class CustomSearchController: UISearchController
    {
        Lazy<CustomSearchBar> _searchBar; 
        
        public CustomSearchController(UIViewController searchResultsController): base(searchResultsController)
        {
        }      

        private void InitSearchBar()
        {
            _searchBar = new Lazy<CustomSearchBar>(() =>
            {
                var result = new CustomSearchBar(CGRect.Empty);
                result.WeakDelegate = this;

                return result;
            });
        }

        public override UISearchBar SearchBar
        {
            get
            {
                if (_searchBar == null)
                    InitSearchBar();

                return _searchBar.Value;
            }
        }
    }
}

File: iOS10/LightOrganApp/CustomSearchController.swift
//
//  CustomSearchController.swift
//  LightOrganApp
//
//  Created by Marcin Kruszyński on 06/05/16.
//  Copyright © 2016 Marcin Kruszyński. All rights reserved.
//

import UIKit

class CustomSearchController: UISearchController, UISearchBarDelegate {
    
    lazy var _searchBar: CustomSearchBar = {
        [unowned self] in
        let result = CustomSearchBar(frame: CGRect.zero)
        result.delegate = self
        
        return result
        }()
    
    override var searchBar: UISearchBar {
        get {
            return _searchBar
        }
    }
}
